packages/next/src/template-renderer/index.tsx

```diff
@@ -30,6 +30,12 @@ export async function TemplateRenderer( {
 	const { editorBlocks, bodyClasses, stylesheets, scripts, scriptModules } =
 		await getTemplateData( pathname || '/' );
 
+	if ( ! editorBlocks ) {
```

Comment:
Our editorBlocks can be `BlockData[] | undefined` `if ( ! editorBlocks )` this condition will check for undefined but what if we are getting an empty array following suggestion will handle that edge case as well.

```suggestion
	if ( ! editorBlocks || ! editorBlocks.length ) {
```